src/Rules/Deprecations/FetchingClassConstOfDeprecatedClassRule.php

```diff
@@ -0,0 +1,58 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Deprecations;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Name;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+
+class FetchingClassConstOfDeprecatedClassRule implements \PHPStan\Rules\Rule
+{
+
+	/**
+	 * @var Broker
+	 */
+	private $broker;
+
+	public function __construct(Broker $broker)
+	{
+		$this->broker = $broker;
+	}
+
+	public function getNodeType(): string
+	{
+		return ClassConstFetch::class;
+	}
+
+	/**
+	 * @param ClassConstFetch $node
+	 * @param \PHPStan\Analyser\Scope $scope
+	 * @return string[] errors
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node->class instanceof Name) {
+			return [];
+		}
+
+		$className = (string) $node->class;
+
+		try {
+			$class = $this->broker->getClass($className);
+		} catch (\PHPStan\Broker\ClassNotFoundException $e) {
+			return [];
+		}
+
+		if (!$class->isDeprecated()) {
+			return [];
+		}
+
+		return [sprintf(
+			'Fetching class constant of deprecated class %s.',
```

Comment:
Which class constant?

Reply:
There's only one class constant 🙂

E.g. `stdClass::class`

Reply:
IMHO this rule works for all accessed class constants...

Reply:
Haha yeah sorry. Apparently I'm confused. But that made me notice that we're totally missing deprecations of constants! I'll fix that too.